File: java/etl/src/main/java/Etl.java
import java.util.HashMap;
import java.util.List;
import java.util.Map;

class Etl {
    Map<String, Integer> transform(Map<Integer, List<String>> old) {

        Map<String, Integer> map = new HashMap<>();

        for (Integer letterValue : old.keySet()) {

            for (String letter : old.get(letterValue)) {
                map.put(letter.toLowerCase(), letterValue);
            }
        }
        return map;
    }
}

File: kotlin/etl/src/main/kotlin/Etl.kt
object ETL {
  fun transform(source: Map<Int, Collection<Char>>) = source.keys
    .fold(mutableMapOf<Char, Int>()) { res, key ->
      for (c in source[key]!!) {
        res.put(c.toLowerCase(), key)
      }
      res
    }
}
